WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -117,18 +117,45 @@
     expect(competition).to be_invalid
   end
 
-  it "requires unapproved competition name is not greater than 32 characters" do
+  it "requires competition name is not greater than 50 characters" do
     competition = FactoryGirl.create :competition
-    competition.showAtAll = false
-    competition.name = "A really long competition name 2016"
+    competition.showAtAll = true
+    competition.name = "A really long competition name that is greater than 50 characters 2016"
     expect(competition).to be_invalid
+    expect(competition.errors.messages[:name]).to eq ["is too long (maximum is 50 characters)"]
   end
 
-  it "competition name can be greater than 32 characters when approved" do
+  it "warns if competition name is greater than 32 characters" do
     competition = FactoryGirl.create :competition
-    competition.showAtAll = true
     competition.name = "A really long competition name 2016"
     expect(competition).to be_valid
+    expect(competition.warnings[:name]).to eq "The competition name is too long (maximum is 32 characters)"
+  end
+
+  it "warns if competition is not visible" do
+    competition = FactoryGirl.create :competition
+    competition.showAtAll = false
+    expect(competition).to be_valid
+    expect(competition.warnings[:invisible]).to eq "This competition is not visible to the public."
+  end
+
+  it "displays info if competition is finished but results aren't uploaded" do
+    competition = FactoryGirl.create :competition
+    competition.start_date = "2016-04-24"
+    competition.end_date = "2016-04-24"
```

Comment:
Also we have a nice way for setting start/end dates, so you can write it as follows:
```ruby
competition = FactoryGirl.create :competition, starts: 1.month.ago
```
This sets the start date to the point in the past. End date will equal to the start date if you don't specify it explicitly, so you can avoid repeating.